heron/ckptmgr/src/java/com/twitter/heron/ckptmgr/CheckpointManagerServer.java

```diff
@@ -0,0 +1,298 @@
+  }
+
+  @Override
+  public void onConnect(SocketChannel channel) {
+    LOG.info("Got a new connection from host:port "
+        + channel.socket().getRemoteSocketAddress());
+  }
+
+  @Override
+  public void onRequest(REQID rid, SocketChannel channel, Message request) {
+    if (request instanceof CheckpointManager.RegisterStMgrRequest) {
+      handleStMgrRegisterRequest(rid, channel, (CheckpointManager.RegisterStMgrRequest) request);
+    } else if (request instanceof CheckpointManager.RegisterTMasterRequest) {
+      handleTMasterRegisterRequest(rid, channel,
+                                   (CheckpointManager.RegisterTMasterRequest) request);
+    } else if (request instanceof CheckpointManager.SaveInstanceStateRequest) {
+      handleSaveInstanceStateRequest(
+          rid, channel, (CheckpointManager.SaveInstanceStateRequest) request);
+    } else if (request instanceof CheckpointManager.GetInstanceStateRequest) {
+      handleGetInstanceStateRequest(
+          rid, channel, (CheckpointManager.GetInstanceStateRequest) request);
+    } else if (request instanceof CheckpointManager.CleanStatefulCheckpointRequest) {
+      handleCleanStatefulCheckpointRequest(
+          rid, channel, (CheckpointManager.CleanStatefulCheckpointRequest) request);
+    } else {
+      LOG.severe("Unknown kind of request: " + request.getClass().getName());
+    }
+  }
+
+  protected void handleCleanStatefulCheckpointRequest(
+      REQID rid,
+      SocketChannel channel,
+      CheckpointManager.CleanStatefulCheckpointRequest request
+  ) {
+    LOG.info("Got a clean request from " + request.toString() + " host:port "
+        + channel.socket().getRemoteSocketAddress());
+
+    boolean deleteAll = request.hasCleanAllCheckpoints() && request.getCleanAllCheckpoints();
+    Common.StatusCode statusCode = Common.StatusCode.OK;
+
+    try {
+      checkpointsBackend.dispose(topologyName,
+                                 request.getOldestCheckpointPreserved(), deleteAll);
+      LOG.info("Dispose checkpoint successful");
+    } catch (StatefulStorageException e) {
+      LOG.info("Dispose checkpoint failed with error " + e);
+      statusCode = Common.StatusCode.NOTOK;
+    }
+
+    CheckpointManager.CleanStatefulCheckpointResponse.Builder responseBuilder =
+        CheckpointManager.CleanStatefulCheckpointResponse.newBuilder();
+    responseBuilder.setStatus(Common.Status.newBuilder().setStatus(statusCode));
+
+    sendResponse(rid, channel, responseBuilder.build());
+  }
+
+  protected void handleTMasterRegisterRequest(
+      REQID rid,
+      SocketChannel channel,
+      CheckpointManager.RegisterTMasterRequest request
+  ) {
+    LOG.info("Got a register request from TMaster host:port "
+        + channel.socket().getRemoteSocketAddress());
+
+    CheckpointManager.RegisterTMasterResponse.Builder responseBuilder =
+        CheckpointManager.RegisterTMasterResponse.newBuilder();
+
+    if (!request.getTopologyName().equals(topologyName)) {
```

Comment:
This assertion logic is copied in this class. Please make a generic assertValidRequest method so we can get code reuse.

How come the assertion of valid topologyName and topologyId only happens in some methods but not all?

(I have a broader question which is why is topologyName/topologyId even passed around at all, since they are know at startup time and are static.)